assets/app/view/game/part/city_slot.rb

```diff
@@ -76,15 +76,23 @@ def reservation
           h(:text, { attrs: attrs }, text)
         end
 
+        def cheater_token(entity)
+          token_abilities = entity.abilities(:token)
+          return nil if token_abilities.empty?
+
+          token_abilities&.cheater
```

Comment:
if token_abilities can be nil, line 81 will crash, if it can't be nil, why do you use &?

Reply:
I rebased to the latest master and the original cheater code now works. I will return this file's code to the previous implementation.